### PR TITLE
[BE] correct the check_pending_webhooks command

### PR DESCRIPTION
**Pull request by poxip**
* fixed the command behavior
* added settings
* added ENV_PREFIX info to the events pending email

### CODE REVIEW

**Comment by poxip** [comment]
FYI The coverage decreased because of a change in migration.

File: README.rst [hunk 1]
```diff
@@ -15,11 +15,22 @@ At the moment the library supports:
...
 
-Add ``aa_stripe.api_urls`` into your url conf.
+Sample ``STRIPE_SETTINGS``:
+
+::
+
+    STRIPE_SETTINGS = {
+        "API_KEY": "api_key",
+        "WEBHOOK_ENDPOINT_SECRET": "somesecret",
+        "USER_MODEL": "myapp.Account"
```
Comment (jacoor) [inline README.rst]: I don't see limit of pending webhooks here. Please add.

File: README.rst [hunk 2]
```diff
@@ -121,12 +132,12 @@ https://stripe.com/docs/api#plans
...
 Be advised. There might be times that Webhooks will not arrive because of some error or arrive in incorrect order. When parsing webhook it is also good to download the refered object to verify it's state.
 
 Stripe has the weird tendency to stop sending webhooks, and they have not fixed it yet on their side. To make sure all events have arrived into your system, the ``check_pending_webhooks`` management command should be run chronically.
-In case there is more pending webhooks than specified in the ``PENDING_EVENTS_THRESHOLD`` variable in settings (default: ``20``), an email to project admins will be sent with ids of the pending events, and also the command will fail raising an exception,
-so if you have some kind of error tracking service configured on your servers (for example: `Sentry <https://sentry.io>`_), you will be notified.
+In case there is more pending webhooks than specified in the ``PENDING_EVENTS_THRESHOLD`` variable in your stripe settings (default: ``20``), an email to project admins will be sent with ids of the pending events, and also the command will fail raising an exception,
```
Comment (jacoor) [inline README.rst]: mentioned here, not in the settings above. Please update settings above.

File: README.rst [hunk 3]
```diff
@@ -121,12 +132,12 @@ https://stripe.com/docs/api#plans
...
+so if you have some kind of error tracking service configured on your servers (for example: `Sentry <https://sentry.io>`_), you will be notified. Also if ``ENV_PREFIX`` is specified in your settings file (not the ``STRIPE_SETTINGS`` configuration), it will be included in the email to admins to indicate on which server the fail occurred.
```
Comment (jacoor) [inline README.rst]: What about domain/site? Django has sites framework and current site should be included in the report.

File: aa_stripe/management/commands/check_pending_webhooks.py [hunk 4]
```diff
@@ -6,19 +6,21 @@
...
+from aa_stripe.settings import stripe_settings
 
 
 class StripePendingWebooksLimitExceeded(Exception):
     def __init__(self, pending_events):
-        self.message = "Pending events limit exceeded, current threshold is {}".format(get_pending_events_threshold)
+        self.message = "Pending events limit exceeded, current threshold is {}".format(
+            stripe_settings.PENDING_EVENTS_THRESHOLD)
         # send email to admins
+        server_env = getattr(settings, "ENV_PREFIX", None)
```
Comment (jacoor) [inline aa_stripe/management/commands/check_pending_webhooks.py]: missing site. Env might be usefull but I'd also add site.

File: aa_stripe/management/commands/check_pending_webhooks.py [hunk 5]
```diff
@@ -27,18 +29,22 @@ class Command(BaseCommand):
...
         pending_events = []
-        last_event = StripeWebhook.objects.latest("created").id
+        last_event = StripeWebhook.objects.last()
+        last_event_id = last_event.id if last_event else None
         while True:
-            event_list = stripe.Event.list(starting_after=last_event, limit=100)  # 100 is the maximum
-            pending_events += event_list["data"]
+            event_list = stripe.Event.list(ending_before=last_event_id, limit=100)  # 100 is the maximum
+            for event in event_list["data"]:
+                if event["pending_webhooks"] > 0:
```
Comment (jacoor) [inline aa_stripe/management/commands/check_pending_webhooks.py]: Please fix.
The story says: "count the amount of events and sent alerts depending on that. The thing is that those are the awaiting events." so clearly about counting not received events. You are checking amount of pending webhooks here for each event not pending webhooks. The list you are receiving here should contain only the events that we not have on our end, so simply counting those should be enough.

File: aa_stripe/management/commands/check_pending_webhooks.py [hunk 6]
```diff
@@ -27,18 +29,22 @@ class Command(BaseCommand):
...
+                    pending_events.append(event)
```
Comment (jacoor) [inline aa_stripe/management/commands/check_pending_webhooks.py]: clearly, this is not pending events amount but amount of pending webhooks for the events

File: aa_stripe/settings.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    "USER_MODEL": settings.AUTH_USER_MODEL
+}
+
+# List of settings that may be in string import notation.
+IMPORT_STRINGS = []
+
+
+class StripeSettings(object):
+    """
+    A settings object, that allows API settings to be accessed as properties.
```
Comment (jacoor) [inline aa_stripe/settings.py]: API?

File: aa_stripe/settings.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+"""
+Settings for aa-stripe are all namespaced in the STRIPE_SETTINGS setting. For example your project's "settings.py" file
+might look like this:
+
+STRIPE_SETTINGS = {
+    "PENDING_EVENTS_THRESHOLD": 20
+}
+
+To simplify overriding those settings they have a flat structure.
```
Comment (jacoor) [inline aa_stripe/settings.py]: Flat structure? please explain.

File: aa_stripe/settings.py [hunk 9]
```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+"""
```
Comment (jacoor) [inline aa_stripe/settings.py]: Usage of this file is wrong. If you look https://github.com/ArabellaTech/roman_api/pull/173#discussion-diff-136361453R4 it is described clearly. Please rework the settings to import from our project, not the other way.

File: aa_stripe/settings.py (hunk 8 above)
Comment (poxip) [reply aa_stripe/settings.py]: Well, this is based on settings from chat_api (which was based on API settings from Rest Framework, right @pkrzyzaniak?). I will change it to have flat settings as you requested in the other PR.

File: README.rst (hunk 3 above)
Comment (poxip) [reply README.rst]: Well, current site cannot be get without a request object, but I will add a "--site" parameter (with default: `Site.object.first()`). I will also add a note about this to README.

**Comment by poxip** [comment]
What was changed:
* rewrote the settings (now everything is configured in project settings - `STRIPE_API_KEY`, etc.)
* corrected counting the number of pending webhooks
* added site to email

**Comment by jacoor** [comment]
This command is still not working as expected.
You need to check your work - here, the command always returned 100 events for me because the ordering is not defined.  I added meta with ordering for the model and it started to work properly.

**Comment by jacoor** [comment]
It worked by a miracle. 
Command: last_event = StripeWebhook.objects.first()
tests: last_webhook = StripeWebhook.objects.last()

**Pull request merged**